### PR TITLE
feat: added coredns patch in terraform template

### PR DESCRIPTION
**Pull request by Al-Pragliola**
needs https://github.com/sighupio/furyctl/pull/327 to be merged

### CODE REVIEW

File: templates/distribution/terraform/coredns.tf.tpl [hunk 1]
```diff
@@ -0,0 +1,34 @@
+locals {
+  patch_data = {
```
Comment (g-iannelli) [inline templates/distribution/terraform/coredns.tf.tpl:2]:
```suggestion
  coredns_scheduling_patch = {
```

File: templates/distribution/terraform/coredns.tf.tpl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+              value = "{{ $value.value }}"
+              effect = "{{ $value.effect }}"
+            },
+            {{- end }}
+          ]
+        }
+      }
+    }
+  }
+}
```
Comment (g-iannelli) [inline templates/distribution/terraform/coredns.tf.tpl:24]:
```suggestion
 coredns_scheduling_patch_as_json = jsonencode(local.patch_data)
}
```

File: templates/distribution/terraform/coredns.tf.tpl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+resource "null_resource" "patch_coredns" {
+  triggers = {
+    run_once = jsonencode(local.patch_data)
+  }
+
+  provisioner "local-exec" {
+    command = "${var.kubectl_path} patch deployment/coredns -n kube-system -p '${jsonencode(local.patch_data)}'"
```
Comment (g-iannelli) [inline templates/distribution/terraform/coredns.tf.tpl:32]:
```suggestion
    command = "${var.kubectl_path} patch deployment/coredns -n kube-system -p '${local.coredns_scheduling_patch_as_json}'"
```

File: templates/distribution/terraform/variables.tf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+variable "kubectl_path" {
+  type        = string
+  description = "The path to the kubectl binary"
+}
```
Comment (g-iannelli) [inline templates/distribution/terraform/variables.tf:4]:
```suggestion
variable "kubectl_path" {
  type        = string
  description = "The path to the kubectl binary. By default, the one present in PATH is used"
  default = "kubectl"
}

```

**Review by g-iannelli — CHANGES_REQUESTED** [review]

**Comment by Al-Pragliola** [comment]
Closing, as it's not relevant anymore.

**Pull request closed without merging**